src/devices/nm-device.c

```diff
@@ -1743,9 +1746,36 @@ get_ip_iface_identifier (NMDevice *self, NMUtilsIPv6IfaceId *out_iid)
 	if (pllink->addr.len > NM_UTILS_HWADDR_LEN_MAX)
 		g_return_val_if_reached (FALSE);
 
+	hwaddr = pllink->addr.data;
+	hwaddr_len = pllink->addr.len;
+
+	if (pllink->type == NM_LINK_TYPE_6LOWPAN) {
+		/* If the underlying IEEE 802.15.4 device has a short address we generate
+		 * a "pseudo 48-bit address" that's to be used in the same fashion as a
+		 * wired Ethernet address. The mechanism is specified in Section 6. of
+		 * RFC 4944 */
+		guint16 pan_id;
+		guint16 short_addr;
+		guint8 pseudo_hwaddr[ETH_ALEN];
+
+		short_addr = nm_platform_wpan_get_short_addr (platform, pllink->parent);
```

Comment:
this not only looks into the platform cache, it make a genl request. I dislike the idea, that in order to generate the interface-identifyer, we need to ask kernel. I guess, that cannot easily be avoided, can it? In the pull-request summary, you alluded to the fact that we cannot cache the value, because we don't get change notifications via rtnetlink.

Reply:
I think fixing this would require kernel change.